File: ui-ngx/src/app/core/http/scheduler.service.ts
///
/// Copyright © 2016-2025 The Thingsboard Authors
///
/// Licensed under the Apache License, Version 2.0 (the "License");
/// you may not use this file except in compliance with the License.
/// You may obtain a copy of the License at
///
///     http://www.apache.org/licenses/LICENSE-2.0
///
/// Unless required by applicable law or agreed to in writing, software
/// distributed under the License is distributed on an "AS IS" BASIS,
/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
/// See the License for the specific language governing permissions and
/// limitations under the License.
///

import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { TranslateService } from '@ngx-translate/core';
import { PageData, PageLink } from '@app/shared/public-api';
import { Observable } from 'rxjs';
import { RequestConfig, defaultHttpOptionsFromConfig } from './http-utils';
import { Scheduler } from '@app/shared/models/scheduler.models';


@Injectable({
    providedIn: 'root'
})
export class SchedulerService {

    constructor(
        private http: HttpClient,
        private translate: TranslateService

    ) { }

    public getTenantScheduler(pageLink: PageLink, config?: RequestConfig): Observable<PageData<Scheduler>> {
        return this.http.get<PageData<Scheduler>>(`/api/tenant/schedulers${pageLink.toQuery()}`,
            defaultHttpOptionsFromConfig(config));
    }

    public saveScheduler(scheduler: Scheduler, config?: RequestConfig): Observable<Scheduler> {
        return this.http.post<Scheduler>('/api/scheduler', scheduler, defaultHttpOptionsFromConfig(config));
    }

    public deleteScheduler(schedulerId: string, config?: RequestConfig) {
        return this.http.delete(`/api/scheduler/${schedulerId}`, defaultHttpOptionsFromConfig(config));
    }
}

File: common/dao-api/src/main/java/org/thingsboard/server/dao/scheduler/SchedulerService.java
/**
 * Copyright © 2016-2025 The Thingsboard Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.thingsboard.server.dao.scheduler;

import org.thingsboard.server.common.data.EntityType;
import org.thingsboard.server.common.data.Role;
import org.thingsboard.server.common.data.SchedulerEvent;
import org.thingsboard.server.common.data.id.*;
import org.thingsboard.server.common.data.page.PageData;
import org.thingsboard.server.common.data.page.PageLink;
import org.thingsboard.server.dao.entity.EntityDaoService;

import java.util.List;
import java.util.Optional;

public interface SchedulerService extends EntityDaoService{

  PageData<SchedulerEvent> findSchedulerByTenantId (TenantId tenantId , PageLink pageLink);
    PageData<SchedulerEvent> findSchedulerByUserId (TenantId tenantId , UserId userId, PageLink pageLink);

    SchedulerEvent saveScheduler (TenantId tenantId , SchedulerEvent schedulerEvent);
  SchedulerEvent findSchedulerById(TenantId tenantId, SchedulerEventId schedulerEventId);
  void deleteScheduler (TenantId tenantId,  SchedulerEventId schedulerEventId);
  List<SchedulerEvent> findAllSchedulerEvent();
  void deleteAllByTenantId( TenantId tenantId);
}
